metadata-ws/src/test/resources/features/accession.feature

```diff
@@ -0,0 +1,68 @@
+Feature: accession object
+
+  Scenario Outline: find different objects by accession and version
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 9606,
+      "name": "Homo Sapiens"
+    }
+    """
+    Then set the URL to TEST_TAXONOMY
+    When I create a test parameterized study with EGAS0001 for accession, 1 for version, test_study for name, false for deprecated, 0 for releaseDay and TEST_TAXONOMY for taxonomy
+    And I request POST /files with JSON payload:
+    """
+    {
+      "accessionVersionId": {
+        "accession": "EGAF0001",
+        "version": 1
+        },
+      "hash": "asd123",
+      "name": "file1",
+      "size": 100,
+      "type": "TSV"
+    }
+    """
+    Then set the URL to TEST_FILE
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 1,
+      "name": "Species1"
+    }
+    """
+    Then set the URL to TEST_TAXONOMY1
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 2,
+      "name": "Species2"
+    }
+    """
+    Then set the URL to TEST_TAXONOMY2
+    When I create a test parameterized sample with EGAN0001 for accession, 1 for version, Sample1 for name and TEST_TAXONOMY1,TEST_TAXONOMY2 for taxonomy
```

Comment:
You need to define scenarios for each testing like accession study, accession files etc

Reply:
ok, then all necessary steps should be repeated for accession study, accession files etc

Reply:
I agree with @selvaebi  here. The secenario here is for testing the searchability of objects through accessionVersionId. That is probabaly one of the most important ways for a client to locate an object. So please make sure that every objects that have `accessionVersionId` can be searched by `accessionVersionId`. That means, in addition to `File`, `Sample` and `Study`, we should also test `Analysis` for this.

Probably there should be step definitions for those four object types which allows creating a default test object or creating a test object with accession and version as arguments. Then you can combine testings on those four different types of object into a "Scenario Outline".

For what it's worth, I don't think the repetition in Cucumber steps is necessarily a bad thing. After all, in Cucumber's context, specifications, tests and documentation are in the **same document**.  They are more like business documents, definitely not code. So we don't treat them in the same way as we treat code in terms of abstraction. I think they should be plain, detailed, easy to understand and repeated wherever needed.

Reply:
I have separated accession validation and accession search. If any comment still applies, i will address

Reply:
Creating a scenario for each object(study, taxonomy, analysis,etc.. ) would be clean and presentable.

Reply:
done